HyperVExtension/test/HyperVExtension/HyperVExtensionTests/Services/HyperVExtensionIntegrationTest.cs

```diff
@@ -131,70 +132,59 @@ public void TestConfigureRequest()
-                else if (progressData.SetState == ConfigurationSetState.WaitingForUserLogon)
-                {
-                    var extensionAdaptiveCard = new Mock<IExtensionAdaptiveCard>();
-                    extensionAdaptiveCard
-                        .Setup(x => x.Update(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>()))
-                        .Returns((string templateJson, string dataJson, string state) => new ProviderOperationResult(ProviderOperationStatus.Success, null, null, null));
-
-                    extensionAdaptiveCard
-                        .Setup(x => x.State)
-                        .Returns("WaitForVmUserLogin");
-
-                    // TODO: figure out how to wait for user's login
-                    progressData.CorrectiveActionCardSession.Initialize(extensionAdaptiveCard.Object);
-                    var op = progressData.CorrectiveActionCardSession.OnAction(@"{ ""Type"": ""Action.Execute"", ""Id"": ""okAction"" }", @"{ }");
-                    await op.AsTask();
-                }
+               var extensionAdaptiveCard = new Mock<IExtensionAdaptiveCard>();
+               extensionAdaptiveCard
+                    .Setup(x => x.Update(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>()))
+                    .Returns((string templateJson, string dataJson, string state) => new ProviderOperationResult(ProviderOperationStatus.Success, null, null, null));
+
+               extensionAdaptiveCard
+                    .Setup(x => x.State)
+                    .Returns("VmCredential");
+
+               credentialsCardSession.Initialize(extensionAdaptiveCard.Object);
+               var op = credentialsCardSession.OnAction(@"{ ""Type"": ""Action.Execute"", ""Id"": ""okAction"" }", @"{ ""id"": ""okAction"", ""UserVal"": """", ""PassVal"": """" }");
+               await op.AsTask();
+            }
+            else if (actionRequired.CorrectiveActionCardSession is WaitForLoginAdaptiveCardSession waitForLoginCardSession)
+            {
+                var extensionAdaptiveCard = new Mock<IExtensionAdaptiveCard>();
+                extensionAdaptiveCard
+                    .Setup(x => x.Update(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>()))
+                    .Returns((string templateJson, string dataJson, string state) => new ProviderOperationResult(ProviderOperationStatus.Success, null, null, null));
+
+                extensionAdaptiveCard
+                    .Setup(x => x.State)
+                    .Returns("WaitForVmUserLogin");
+
+                // TODO: figure out how to wait for user's login
+                waitForLoginCardSession.Initialize(extensionAdaptiveCard.Object);
+                var op = waitForLoginCardSession.OnAction(@"{ ""Type"": ""Action.Execute"", ""Id"": ""okAction"" }", @"{ }");
+                await op.AsTask();
             }
         };
 
-        var result = operation.CompletionStatus;
+        var result = await operation.StartAsync();

```

Comment:
In the future we should have a time limit for this wait. If something went wrong and we never received operation completion from the VM we'll wait forever.

Reply:
thats fair, I guess we can update it the same time we remove the async Task